packages/cli/README.md

```diff
@@ -86,6 +87,50 @@ artifacts/           # Hierarchical output
     Token/
 ```
 
+### Structure Mismatch Detection
+
+The compiler tracks which structure type was used via a `manifest.json` file in the output directory. When switching between flattened and hierarchical structures:
+
+- **Interactive mode (TTY):** Prompts for confirmation before deleting existing artifacts
+- **Non-interactive mode (CI/CD):** Requires `--force` flag to proceed
+
+```bash
+$ compact-compiler --hierarchical
+
+⚠ [COMPILE] Existing artifacts use "flattened" structure.
+⚠ [COMPILE] You are compiling with "hierarchical" structure.
+? Delete existing artifacts and recompile? (y/N)
+```
+
+To skip the prompt in scripts or CI/CD:
+
+```bash
+compact-compiler --hierarchical --force
+```
+
+### Manifest File
+
+The compiler generates a `manifest.json` in the output directory with build metadata:
+
+```json
+{
+  "structure": "hierarchical",
+  "compactcVersion": "0.26.0",
+  "compactToolVersion": "0.3.0",
+  "createdAt": "2025-12-11T10:35:09.916Z",
+  "buildDuration": 2445,
```

Comment:
I'm guessing this is in seconds but that should be made clear in the documentation